presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java

```diff
@@ -239,6 +246,43 @@ private boolean nextRow(MessageAndOffset messageAndOffset)
             return true; // Advanced successfully.
         }
 
+        private void openFetchRequest()
+        {
+            if (messageAndOffsetIterator == null) {
+                log.info("Fetching %d bytes from partition %d @offset %d (%d - %d) -- %d messages read so far",
+                        fetchSize, split.getPartitionId(), cursorOffset, split.getStart(), split.getEnd(), totalMessages);
+                cursorOffset += fetchedSize;
+                FetchRequest req = new FetchRequest(split.getTopicName(), split.getPartitionId(), cursorOffset, fetchSize);
```

Comment:
This commit should be what is on master IMO. Then we can make our own modifications as PRs on top of that. This would make it easier to differentiate the fork from our edits.